airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -677,7 +677,7 @@ def create_simple_retriever(self, model: SimpleRetrieverModel, config: Config, *
         requester = self._create_component_from_model(model=model.requester, config=config)
         record_selector = self._create_component_from_model(model=model.record_selector, config=config)
         paginator = (
-            self._create_component_from_model(model=model.paginator, config=config, url_base=model.requester.url_base)
+            self.create_default_paginator(model=model.paginator, url_base=requester.get_url_base(), config=config)
```

Comment:
also something worth noting that we should probably be using the model to derive the value of `url_base` because when we do `requester.get_url_base()` we will evaluate the interpolated value. We should really be storing the template `{{ config['url'] }}` instead and then we perform the evaluation at runtime during a sync.

It is a slight nuance, but a paradigm we should try to maintain as much as possible for a simpler mental model of where in the process we perform interpolation

Reply:
Those are all good points and this was "legacy" change when I was doing investigation as to how to implement the solution. I reverted this to `self._create_component_from_model(model=model.paginator, config=config, url_base=model.requester.url_base)` like it was before. Thanks for pointing that out!

Reply:
Note regarding this: if someone implements a custom requester, the attribute `model.url_base` is not guaranteed. Should we do anything about his?

Reply:
Good callout, I think we can probably just fill it w/ either an empty string. That `url_base` is used by the paginator to remove the base from the complete URL. So an empty string would still work, even though it might not be quite what the developer intends if it is empty.